pkg/p2p/libp2p/libp2p.go

```diff
@@ -685,17 +816,31 @@ func (s *Service) Addresses() (addresses []ma.Multiaddr, err error) {
 		if err != nil {
 			return nil, err
 		}
-
 		addresses = append(addresses, a)
 	}
 	if s.natAddrResolver != nil && len(addresses) > 0 {
 		a, err := s.natAddrResolver.Resolve(addresses[0])
 		if err != nil {
 			return nil, err
 		}
-		addresses = append(addresses, a)
+		// Remove /p2p/<ID> to append /ws before it
```

Comment:
This is done only for the first address (line 822 addresses[0]), maybe this is not even correct on master. All addresses should be resolved. I am also not sure why `/ws` address has to be appended manually. Should it already be in s.host.Addrs()?

Reply:
We should iterate on all addresses and do the resolve on all of them, and if the address is a wss address (has the ws transport and sni) to use a dedicated (new instance) of the reolver that for wss (that will use the nat-wss-addr if configured).